src/main/java/io/toast/tk/maven/plugin/RunScriptsMojo.java

```diff
@@ -0,0 +1,167 @@
+        requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME)
+public class RunScriptsMojo extends AbstractMojo {
+
+    @Parameter(required = false, alias = "pluginsDirectory")
+    private String pluginDir;
+
+    @Parameter(required = true, alias = "outputDirectory")
+    private String outputDirectory;
+
+    @Parameter(required = false, alias = "scripts")
+    private FileSet[] scripts;
+
+    @Parameter(required = true, defaultValue = "${project}", readonly = true)
+    private MavenProject project;
+
+    private TestPageRunner testPageRunner;
+    private Module[] pluginModules;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        try {
+            try {
+                extendPluginClassPath();
+            } catch (DependencyResolutionRequiredException e) {
+                throw new MojoExecutionException(e.getMessage(), e);
+            }
+
+            initReportOutputDir();
+
+            initPluginModules();
+
+            for (int i =0; i < scripts.length; i++){
+                List<File> files = toFileList(scripts[i]);
+                final List<ITestPage> testScripts = getScripts(files);
+                executeScripts(testScripts);
+            }
+        } catch (IOException e) {
+            getLog().error(e);
+        }
+    }
+
+    private void initPluginModules() {
+        PluginLoader loader = new PluginLoader(this.pluginDir);
+        this.pluginModules = pluginDir == null ? new Module[]{} : loader.collectGuiceModules(loader.loadPlugins(IAgentPlugin.class.getClassLoader()));
+    }
+
+    private void initReportOutputDir() throws IOException {
+        File file = new File(this.outputDirectory);
+        if(!file.exists()){
+            file.mkdir();
+        } else if(!file.isDirectory()){
+            throw new IOException(this.outputDirectory + "isn't a valid directory !");
+        }
+    }
+
+    private List<File> toFileList(FileSet scriptSet) throws IOException {
+        File directory = new File(scriptSet.getDirectory());
+        String includes = toString(scriptSet.getIncludes());
+        String excludes = toString(scriptSet.getExcludes());
+        return FileUtils.getFiles(directory, includes, excludes);
+    }
+
+    private static String toString(List<String> strings) {
+        StringBuilder sb = new StringBuilder();
+        for (String string : strings) {
+            if (sb.length() > 0)
+                sb.append(", ");
+            sb.append(string);
+        }
+        return sb.toString();
+    }
+
+    private List<ITestPage> getScripts(List<File> files) {
+        List<ITestPage> testScripts = new ArrayList<>();
+        TestParser parser = new TestParser();
+        files.forEach(p -> {
+            try{
+                List<String> scriptLines = FileHelper.getScript(new FileInputStream(p));
+                ITestPage testScript = parser.parse(scriptLines, p.getName().toString());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') "getName" returns a string, there's no need to call "toString()". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1858)